### PR TITLE
Improve text input

### PR DESCRIPTION
**Pull request by felipemanga**
Closes LibreSprite/LibreSprite-proposals#36
It seems the font library also supports otf files, so I let it load them as well while I was at it.

### CODE REVIEW

File: src/ui/entry.cpp [hunk 1]
```diff
@@ -200,7 +202,20 @@ bool Entry::onProcessMessage(Message* msg)
...
       break;
 
-    case kFocusEnterMessage:
+    case kFocusEnterMessage: {
+      m_got_focus_message = true;
+      gfx::Rect rect = bounds();
+      View* view = View::getView(this);
+      if (view) {
+        rect = view->viewportBounds();
+      }
```
Comment (Zughy) [inline src/ui/entry.cpp:211]: how about
```cpp
View* view = View::getView(this);
gfx::Rect = view ? view->viewportBounds() : bounds();
```
?

File: data/widgets/paste_text.xml [hunk 2]
```diff
@@ -1,10 +1,11 @@
 <!-- LibreSprite -->
 <!-- Copyright (C) 2015 by David Capello -->
+<!-- Copyright (C) 2021 LibreSprite contributors -->
 <gui>
 <window text="Insert Text" id="paste_text">
   <grid columns="2">
     <label text="Text:" />
-    <entry expansive="true" maxsize="256" id="user_text" magnet="true" cell_align="horizontal" />
+    <entry expansive="true" maxsize="256" id="user_text" magnet="true" cell_align="horizontal" height="20" width="200"/>
```
Comment (Zughy) [inline data/widgets/paste_text.xml:8]: nitpicking: missing space between `"200"` and `/>`

File: src/app/ui/font_popup.cpp [hunk 3]
```diff
@@ -196,7 +197,8 @@ FontPopup::FontPopup()
 
   // Create one FontItem for each font
   for (auto& file : files) {
-    if (base::string_to_lower(base::get_file_extension(file)) == "ttf")
+    auto ext = base::string_to_lower(base::get_file_extension(file));
+    if (ext == "ttf" || ext == "otf")
```
Comment (Zughy) [inline src/app/ui/font_popup.cpp:201]: nitpicking: is `ext` the common short word for `extension`? I thought about `external` at a first glance, so I'm wondering if it makes more sense to just write the word in full

**Comment by Zughy** [comment]
This enhancement is amazing, just tested

File: src/app/ui/font_popup.cpp (hunk 3 above)
Comment (felipemanga) [reply src/app/ui/font_popup.cpp:201]: Shortening `extension` into `ext` is pretty common throughout the codebase, although it does use the non-abbreviated form as well. :man_shrugging:

Comment (Zughy) [reply src/app/ui/font_popup.cpp:201]: ..mm.

**Review by Zughy — APPROVED** [review]

**Pull request merged**